### PR TITLE
pkg/obfuscate: downgrade easyjson

### PR DESCRIPTION
**Pull request by Taiki-San**
### What does this PR do?

This PR downgrades the version of easyjson used for the serialisation/deserialisation within the obfuscator.

### Motivation

Resolves conflicts with internal projects.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/obfuscate/obfuscate_easyjson.go [hunk 1]
```diff
@@ -61,31 +60,36 @@ func easyjson4ef41860EncodeGithubComDataDogDatadogAgentPkgObfuscate(out *jwriter
...
-		const prefix string = ",\"Cache\":"
-		out.RawString(prefix)
-		out.Bool(bool(in.Cache))
+	first = false
+	out.RawString("\"DollarQuotedFunc\":")
+	out.Bool(bool(in.DollarQuotedFunc))
+	if !first {
+		out.RawByte(',')
 	}
+	first = false
```
Comment (Taiki-San) [inline pkg/obfuscate/obfuscate_easyjson.go:90]:
```suggestion
```
Avoid a CI check

**Review by alexbarksdale — APPROVED** [review]

File: pkg/obfuscate/sql_tokenizer.go [hunk 2]
```diff
@@ -205,7 +205,7 @@ func (tkn *SQLTokenizer) Scan() (TokenKind, []byte) {
 	if tkn.lastChar == 0 {
```
Comment (gbbr) [inline pkg/obfuscate/sql_tokenizer.go:205]: Why are the changes in this file present?

**Review by gbbr — APPROVED** [review]

File: pkg/obfuscate/go.mod [hunk 3]
```diff
@@ -6,6 +6,6 @@ require (
 	github.com/DataDog/datadog-go v4.8.2+incompatible
 	github.com/Microsoft/go-winio v0.5.1 // indirect
 	github.com/dgraph-io/ristretto v0.1.0
-	github.com/mailru/easyjson v0.7.7
+	github.com/mailru/easyjson v0.0.0-20170227080219-44f6bc771e3c
```
Comment (gbbr) [inline pkg/obfuscate/go.mod:9]: Why are we not using a proper version? Why this pseudo-version?

File: pkg/obfuscate/sql_tokenizer.go (hunk 2 above)
Comment (Taiki-San) [reply pkg/obfuscate/sql_tokenizer.go]: Extracted those changes into https://github.com/DataDog/datadog-agent/pull/10129

File: pkg/obfuscate/go.mod (hunk 3 above)
Comment (Taiki-San) [reply pkg/obfuscate/go.mod:9]: This matches the internal project we're targeting.
Unfortunately, the commit precede by two years the earlier published version (v0.7.0, September 10th 2019).
Between the complexity of an upgrade on the other side and the limited scope of the downgrade here, it felt like this was the quickest and least risky change.

Comment (Taiki-San) [reply pkg/obfuscate/go.mod]: I updated to v0.7.0 and it doesn't appear to break anything.
I agree it's much cleaner than tagging a random commit from 2017 :)

**Pull request merged**